packages/terra-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -460,63 +318,25 @@ function WorklistDataGrid(props) {
     }
   }, [onRowSelect, rows]);
 
-  const handleMoveCellFocus = (fromCell, toCell) => {
-    // Obtain coordinate rectangles for grid container, column header, and new cell selection
-    const gridContainerRect = gridContainerRef.current.getBoundingClientRect();
-    const columnHeaderRect = grid.current.rows[0].cells[toCell.col].getBoundingClientRect();
-    const nextCellRect = grid.current.rows[toCell.row].cells[toCell.col].getBoundingClientRect();
-
-    // Calculate horizontal scroll offset for right boundary
-    if (nextCellRect.right > gridContainerRect.right) {
-      gridContainerRef.current.scrollBy(nextCellRect.right - gridContainerRect.right, 0);
-    } else {
-      // Calculate horizontal scroll offset for left boundary
-      let scrollOffsetX = 0;
-      if (pinnedColumnOffsets.length > 0) {
-        if (toCell.col > pinnedColumnOffsets.length - 1) {
-          const lastPinnedColumnRect = grid.current.rows[toCell.row].cells[pinnedColumnOffsets.length - 1].getBoundingClientRect();
-          scrollOffsetX = nextCellRect.left - lastPinnedColumnRect.right;
-        }
-      } else {
-        scrollOffsetX = nextCellRect.left - gridContainerRect.left;
-      }
-
-      if (scrollOffsetX < 0) {
-        gridContainerRef.current.scrollBy(scrollOffsetX, 0);
+  const onRangeSelection = useCallback((rowIndex, columnIndex, direction) => {
```

Comment:
You should add method documentation here to ensure that the logic is easier to maintain.